lib/cookie.js

```diff
@@ -0,0 +1,81 @@
+/*
+ * derived from jQuery Cookie Plugin v1.4.1
+ * https://github.com/carhartl/jquery-cookies
+ */
+
+ // needed to communicate/translate between JSONWire cookies and regular JS cookies 
+
+import _ from 'lodash';
+import { getLogger } from 'appium-logger';
+
+
+export default class Cookie {
+
+  constructor (cookieString) {
+    this.cookieString = cookieString;
+    this.log = getLogger('mymodule');
+  }
+ 
+  // parses the value if needed and converts the value if a converter is provided
+  convertCookie (value, converter) {
+    let parsedValue;
+
+    if (value.indexOf('"') === 0) {
+      value = value.slice(1, -1).replace(/\\"/g, '"').replace(/\\\\/g, '\\');
+    }
+    try {
+      parsedValue = decodeURIComponent(value.replace(/\+/g, ' '));
+    } catch (e) {
+      this.log.warn(e);
+    }
+    
+    return converter ? converter(parsedValue) : parsedValue;
+  }
+
+  // takes arguments given and creates a JS Cookie 
+  writeCookie (key, value, options) { 
+    if (arguments.length > 1 && !_.isFunction(value)) {
```

Comment:
Shouldn't it check if a converter is provided? If a converter was not provided it just returns the parsedValue, but if a converter IS provided then it parses it then proceeds to convert it using the converter as well. If the check isn't here it has to be in readCookie anyways